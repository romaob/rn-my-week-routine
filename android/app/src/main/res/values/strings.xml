<resources>
    <string name="app_name">MyWeekRoutine</string>
</resources>
