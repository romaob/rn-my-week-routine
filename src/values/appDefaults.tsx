export enum APP_KEYS {
  LANGUAGE = '@myweekroutine.language',
  ALERTS_ENABLED = '@myweekroutine.alertsEnabled',
  ONBOARDING = '@myweekroutine.onboarding',
  EVENTS = '@myweekroutine.events',
}

export interface Event {
  id: string;
  name: string;
  indexes: number[];
  description: string;
  startAt: string;
  endAt: string;
  alertEnabled: boolean;
  alertSent: boolean;
  alertConfirmed: boolean;
  added: string;
  updated: string;
}
