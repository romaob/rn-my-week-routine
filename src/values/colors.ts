// App color palette
export const colors = {
  light: {
    primary: '#3F51B5',
    secondary: '#536DFE',
    accent: '#FF4081',
    danger: '#f44336',
    warning: '#ff9800',
    success: '#4caf50',
    info: '#2196f3',
    background: '#ffffff',
    cardBackground: '#ffffff',
    disabled: '#bdbdbd',
    text: '#000000',
    textContrast: '#ffffff',
    textSecondary: '#757575',
    textDisabled: '#bdbdbd',
  },
};
