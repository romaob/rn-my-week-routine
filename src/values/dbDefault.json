{
    "language": "en",
    "alertsEnabled": false,
    "onboarding": false,
    "events": []
}