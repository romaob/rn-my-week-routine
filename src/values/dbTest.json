{
    "language": "en",
    "alertsEnabled": true,
    "onboarding": false,
    "events": [
        {
            "id": 1,
            "name": "Event 1",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 1 description",
            "startAt": "00:00:00",
            "endAt": "01:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 2,
            "name": "Event 2",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 2 description",
            "startAt": "01:00:00",
            "endAt": "02:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 3,
            "name": "Event 3",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 3 description",
            "startAt": "02:00:00",
            "endAt": "03:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 4,
            "name": "Event 4",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 4 description",
            "startAt": "03:00:00",
            "endAt": "04:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 5,
            "name": "Event 5",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 5 description",
            "startAt": "04:00:00",
            "endAt": "05:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        }
    ]
}