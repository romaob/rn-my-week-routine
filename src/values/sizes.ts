//Options for font sizes, padding, margin, etc.
export const sizes = {
  // Font sizes
  font: {
    sm: 12,
    md: 14,
    md2: 18,
    lg: 24,
    xl: 32,
  },
  // Padding sizes
  padding: {
    sm: 8,
    md: 16,
    lg: 24,
  },
  // Margin sizes
  margin: {
    sm: 8,
    md: 16,
    lg: 24,
  },
  // App values
  appValues: {
    timeSlotHeight: 30,
  },
};
