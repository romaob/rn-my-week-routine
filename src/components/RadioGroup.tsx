import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {sizes} from '../values/sizes';
import Label from './Label';
import RadioButton from './RadioButton';

export interface RadioGroupProps {
  label?: string;
  size?: number;
  itemSelected: string;
  items: string[];
  labels?: string[];
  onPress: (item: string) => void;
}

export default function RadioGroup({
  label,
  size = sizes.font.md,
  itemSelected,
  items,
  labels,
  onPress,
}: RadioGroupProps): JSX.Element {
  return (
    <View style={styles.container}>
      {label && <Label text={label} />}
      {items.map((item, index) => (
        <RadioButton
          key={item}
          label={labels ? labels[index] : item}
          size={size}
          selected={itemSelected === item}
          onPress={() => onPress(item)}
        />
      ))}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    justifyContent: 'center',
    gap: sizes.padding.md,
    paddingVertical: sizes.padding.md,
  },
});
