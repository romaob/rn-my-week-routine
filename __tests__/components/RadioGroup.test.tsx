import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import RadioGroup from '../../src/components/RadioGroup';
import {colors} from '../../src/values/colors';

const items = ['item1', 'item2', 'item3'];
const labels = ['label1', 'label2', 'label3'];
const onPress = jest.fn();
const label = 'label';
const size = 20;

function testGroup(result: any, selected: number, labels: string[]) {
  expect(result).toBeDefined();
  expect(result.getByTestId('radioGroup')).toBeDefined();
  const resultItems = result.getAllByTestId('radioButton');
  expect(resultItems.length).toBe(items.length);
  labels.forEach((itemLabel: string) => {
    expect(result.getByText(itemLabel)).toBeDefined();
  });
  const radioCircleInnerItems = result.getAllByTestId('radioButtonCircleInner');
  radioCircleInnerItems.forEach((item: any, index: number) => {
    expect(item).toBeDefined();
    expect(item.props.style.backgroundColor).toBe(
      selected === index ? colors.light.accent : 'transparent',
    );
    expect(item.props.style.width).toBe(size);
  });
}

describe('RadioGroup render tests', () => {
  it('should render the radio group with label', () => {
    const result = render(
      <RadioGroup
        label={label}
        itemSelected={items[1]}
        labels={labels}
        items={items}
        onPress={onPress}
        size={size}
      />,
    );
    expect(result).toMatchSnapshot();
    testGroup(result, 1, labels);
    expect(result.getByText(label)).toBeDefined();
    expect(result.getByTestId('radioGroupLabel')).toBeDefined();
    expect(
      result
        .getByTestId('radioGroupLabel')
        .props.children.some((child: any) => child === label),
    ).toBeTruthy();
  });

  it('should render the radio group without label', () => {
    const result = render(
      <RadioGroup
        itemSelected={items[1]}
        labels={labels}
        items={items}
        onPress={onPress}
        size={size}
      />,
    );
    expect(result).toMatchSnapshot();
    testGroup(result, 1, labels);
    expect(result.queryByTestId('radioGroupLabel')).toBeNull();
    expect(result.queryByText(label)).toBeNull();
  });

  it('should render the radio group without selection', () => {
    const result = render(
      <RadioGroup
        itemSelected={''}
        labels={labels}
        items={items}
        onPress={onPress}
        size={size}
      />,
    );
    expect(result).toMatchSnapshot();
    testGroup(result, -1, labels);
  });

  it('should render the radio group using the values when not having labels', () => {
    const result = render(
      <RadioGroup
        itemSelected={items[1]}
        items={items}
        onPress={onPress}
        size={size}
      />,
    );
    expect(result).toMatchSnapshot();
    testGroup(result, 1, items);
  });
});

describe('RadioGroup onPress tests', () => {
  it('should call the callback function with the right value when pressing a radio button', () => {
    const result = render(
      <RadioGroup
        itemSelected={items[1]}
        labels={labels}
        items={items}
        onPress={onPress}
        size={size}
      />,
    );
    expect(result).toMatchSnapshot();
    testGroup(result, 1, labels);
    expect(onPress).toHaveBeenCalledTimes(0);
    fireEvent.press(result.getAllByTestId('radioButton')[0]);
    expect(onPress).toHaveBeenCalledTimes(1);
    expect(onPress).toHaveBeenCalledWith(items[0]);
  });
});
