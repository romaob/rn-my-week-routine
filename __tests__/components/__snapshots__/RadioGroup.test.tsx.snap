// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup onPress tests should call the callback function with the right value when pressing a radio button 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 16,
      "justifyContent": "center",
      "paddingVertical": 16,
    }
  }
  testID="radioGroup"
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={13}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label1
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={14}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "#d81e8b",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label2
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={15}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label3
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`RadioGroup render tests should render the radio group using the values when not having labels 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 16,
      "justifyContent": "center",
      "paddingVertical": 16,
    }
  }
  testID="radioGroup"
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={10}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        item1
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={11}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "#d81e8b",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        item2
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={12}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        item3
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`RadioGroup render tests should render the radio group with label 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 16,
      "justifyContent": "center",
      "paddingVertical": 16,
    }
  }
  testID="radioGroup"
>
  <Text
    style={
      {
        "color": "#000000",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 14,
      }
    }
    testID="radioGroupLabel"
  >
    label
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label1
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "#d81e8b",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label2
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={3}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label3
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`RadioGroup render tests should render the radio group without label 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 16,
      "justifyContent": "center",
      "paddingVertical": 16,
    }
  }
  testID="radioGroup"
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={4}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label1
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={5}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "#d81e8b",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label2
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={6}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label3
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`RadioGroup render tests should render the radio group without selection 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 16,
      "justifyContent": "center",
      "paddingVertical": 16,
    }
  }
  testID="radioGroup"
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={7}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label1
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={8}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label2
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={9}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="radioButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderColor": "#d81e8b",
            "borderRadius": 999,
            "borderWidth": 2,
            "display": "flex",
            "padding": 5,
          }
        }
        testID="radioButtonCircle"
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "height": 20,
              "width": 20,
            }
          }
          testID="radioButtonCircleInner"
        />
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
        testID="radioButtonLabel"
      >
        label3
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
